### PR TITLE
Completed task

### PR DESCRIPTION
**Pull request by olgamorozova1**
Task 2

### CODE REVIEW

File: testng.xml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+                    <include name="getTotalPriceTestForAllItems"/>
+                    <exclude name="deleteRealItemTest"/>
+                    <exclude name="deleteVirtualItemTest"/>
+                    <include name="deleteRealItemAndCheckTotalPriceTest"/>
+                    <include name="deleteVirtualItemAndCheckTotalPriceTest"/>
+                </methods>
+            </class>
+        </classes>
+    </test>
+</suite>
```
Comment (KirillKarpov) [inline testng.xml:17]: The good rule is to have understandable naming for files. Especially when it is testng xml suites. 
Testng, testng1 and etc. don't look clear.
You can names from from xml file such as ExceptionTestWithDifferentParameters, SmokeTestOfCartClass and etc.

File: testng3.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE suite SYSTEM "https://testng.org/testng-1.0.dtd" >
+<suite name="ExceptionTestWithDifferentParameters" verbose="1">
+    <test name="nonexistentFileTest">
+        <parameter name="path" value="src\main\resources\test-cart.json"/>
```
Comment (KirillKarpov) [inline testng3.xml:5]: As I mentioned before , if you want remove dependency on operation system, slash should be used instead of backslash.

**Review by KirillKarpov — CHANGES_REQUESTED** [review]

**Comment by KirillKarpov** [comment]
@Parameter looks good as solution for 1 task (.. Exception test with >= 5 datasets).
I would like to ask you also write exception test with >= 5 datasets using @DataProvider from testNG

File: src/test/java/parser/JsonParserTestTestNG.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Test(enabled = false, groups = {"functest"})
+    public void testExceptionMessage() {
+        String path = "src\\main\\resources\\test-cart.json";
+        Throwable exception = Assert.expectThrows(NoSuchFileException.class, () -> parser.readFromFile(new File(path)));
+        Assert.assertEquals(("File " + path + ".json not found!"), exception.getMessage());
+    }
+
+    @Test(groups = {"functest"})
+    @Parameters({"path"})
```
Comment (KirillKarpov) [inline src/test/java/parser/JsonParserTestTestNG.java:86]: @Parameter looks good as solution for 1 task (.. Exception test with >= 5 datasets).
I would like to ask you to write exception test with >= 5 datasets using @DataProvider from testNG

**Review by KirillKarpov — CHANGES_REQUESTED** [review]

File: src/test/java/parser/JsonParserTestTestNG.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    private Cart actualCart;
+    private File file;
+
+    @BeforeMethod(groups = {"checkintest"})
+    void createCartAndJSONFile() {
+        expectedCart = new Cart("olga-cart");
+        file = new File("src/main/resources/olga-cart.json");
+    }
+
+    @Test(groups = {"checkintest"})
```
[the remark this answers is not in the document]
Comment (olgamorozova1) [reply src/test/java/parser/JsonParserTestTestNG.java]: Added new Class testGroups with group names, replaced group names in annotations.

File: src/test/java/parser/JsonParserTestTestNG.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        actualCart = parser.readFromFile(file);
+        SoftAssert softAssertion = new SoftAssert();
+        softAssertion.assertEquals(actualCart.getCartName(), expectedCart.getCartName());
+        softAssertion.assertEquals(actualCart.getTotalPrice(), expectedCart.getTotalPrice());
+        softAssertion.assertAll();
+    }
+
+    @Test(enabled = false, groups = {"functest"})
+    public void testExceptionMessage() {
+        String path = "src\\main\\resources\\test-cart.json";
```
[the remark this answers is not in the document]
Comment (olgamorozova1) [reply src/test/java/parser/JsonParserTestTestNG.java]: Fixed.

File: testng.xml (hunk 1 above)
Comment (olgamorozova1) [reply testng.xml]: Renamed.

File: testng3.xml (hunk 2 above)
Comment (olgamorozova1) [reply testng3.xml]: Fixed.

File: src/test/java/parser/JsonParserTestTestNG.java (hunk 3 above)
Comment (olgamorozova1) [reply src/test/java/parser/JsonParserTestTestNG.java:87]: Added new Exception test with data provider annotation.

**Review by KirillKarpov — APPROVED** [review]

**Pull request merged**